massa-protocol-worker/src/handlers/peer_handler/mod.rs

```diff
@@ -630,3 +624,80 @@ impl InitConnectionHandler<PeerId, Context, MessagesHandler> for MassaHandshake
         Ok(())
     }
 }
+
+#[cfg(all(test, feature = "testing"))]
+mod tests {
+    use std::{collections::HashMap, ops::Deref, sync::Arc};
+
+    use massa_channel::MassaChannel;
+    use massa_protocol_exports::ProtocolConfig;
+    use massa_serialization::U64VarIntDeserializer;
+    use massa_signature::KeyPair;
+    use parking_lot::RwLock;
+    use peernet::{peer::InitConnectionHandler, transports::endpoint::Endpoint};
+
+    use crate::{context::Context, messages::MessagesHandler};
+
+    use super::models::PeerDB;
+
+    #[test]
+    fn test_handshake_working_behaviour() {
```

Comment:
Can we add more tests with invalid handshake + handshake that takes too much time here + handshake with close connection in the middle ?

Reply:
I have added invliad handshake and disconnect. "Takes too much time" isn't testable because it's managed by the timeout on the socket directly in peernet